### PR TITLE
feat: [CDS-56452]:config-file-github

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
Added option to select github store in config files, for TAS/SSH/WINRM

#### Screenshots
Screenshot 2023-04-03 at 17 46 31

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFileGitCoreSection/ConfigFileGitCoreSection.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  isReadonly?: boolean
+  showIdentifierField?: boolean
+  filePathFieldWidth?: number
+}
+
+const getAccountUrl = (prevStepData?: ConnectorConfigDTO): string => {
+  return prevStepData?.connectorRef ? prevStepData?.connectorRef?.connector?.spec?.url : prevStepData?.url
+}
+
+export function ConfigFileGitCoreSection(
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFileGitCoreSection/ConfigFileGitCoreSection.tsx:32]: Most of the code here is duplicated. We should plan on reusing existing code.

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx [hunk 2]
```diff
@@ -207,6 +211,23 @@ function ConfigFilesListView({
           />
         )
         break
+      case ConfigFilesToConnectorMap.Git:
+      case ConfigFilesToConnectorMap.Gitlab:
+      case ConfigFilesToConnectorMap.Bitbucket:
+      case ConfigFilesToConnectorMap.Github:
+        configDetailStep = (
+          <GitConfigStep
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx:219]: <CommonManifestDetails /> could have been used with little refactoring in this component.

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/StepUtils.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
+import { isEmpty } from 'lodash-es'
+import { getScopeFromValue } from '@common/components/EntityReference/EntityReference'
+import { Scope } from '@common/interfaces/SecretsInterface'
+import { ConfigFilesMap } from '../../ConfigFilesHelper'
+
+export enum GitRepoName {
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/StepUtils.ts:7]: If any of these are duplicated then let's remove and use the one that are existing in other files.

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/__tests__/GitConfigStore.test.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  handleSubmit,
+  listOfConfigFiles: [],
+  configFileIndex: 0
+}
+
+function WrapperComponent(props: any): JSX.Element {
+  const { initialErrors } = props || {}
+  return (
+    <TestWrapper>
+      <Formik
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/__tests__/GitConfigStore.test.tsx:75]: Formik is not needed here. The component itself has Formik.

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/__tests__/GitConfigStore.test.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        initialErrors={initialErrors}
+        initialValues={{ valuesPath: ['account:/vit', 'account:/vit1'] }}
+        onSubmit={() => undefined}
+        formName="TestWrapper"
+      >
+        {formikProps => (
+          <FormikForm>
+            <GitConfigStep {...formikProps} {...props} />
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/__tests__/GitConfigStore.test.tsx:83]: File name is GitConfigStore while we are testing GitConfigStep. One of them needs a change.

**Review by chetanvekariya-harness — APPROVED** [review]
Remove empty file - src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFileGitCoreSection/utils.ts

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger cypress-cv

**Comment by litvinovstas** [comment]
retrigger jest

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**